templates/PlayFab_InstanceApi.cpp.ejs

```diff
@@ -140,39 +158,21 @@ namespace PlayFab
     // Private PlayFabClientInstanceAPI specific
     bool PlayFabClientInstanceAPI::IsClientLoggedIn()
     {
-        return !this->GetOrCreateAuthenticationContext()->clientSessionTicket.empty();
+        return !this->m_context->clientSessionTicket.empty();
     }
 
-    void PlayFabClientInstanceAPI::MultiStepClientLogin(bool needsAttribution)
+    void PlayFabClientInstanceAPI::MultiStepClientLogin(std::shared_ptr<PlayFabAuthenticationContext> context, bool needsAttribution)
     {
-        std::shared_ptr<PlayFab::PlayFabApiSettings> apiSettings = this->GetSettings();
-        if (apiSettings == nullptr)
+        if (needsAttribution && !context->disableAdvertising && context->advertisingIdType.length() > 0 && context->advertisingIdValue.length() > 0)
         {
-            if (needsAttribution && !PlayFabSettings::disableAdvertising && PlayFabSettings::advertisingIdType.length() > 0 && PlayFabSettings::advertisingIdValue.length() > 0)
-            {
-                AttributeInstallRequest request;
-                if (PlayFabSettings::advertisingIdType == PlayFabSettings::AD_TYPE_IDFA)
-                    request.Idfa = PlayFabSettings::advertisingIdValue;
-                else if (PlayFabSettings::advertisingIdType == PlayFabSettings::AD_TYPE_ANDROID_ID)
-                    request.Adid = PlayFabSettings::advertisingIdValue;
-                else
-                    return;
-                AttributeInstall(request, nullptr, nullptr);
-            }
-        }
-        else
-        {
-            if (needsAttribution && !apiSettings->disableAdvertising && apiSettings->advertisingIdType.length() > 0 && apiSettings->advertisingIdValue.length() > 0)
-            {
-                AttributeInstallRequest request;
-                if (apiSettings->advertisingIdType == PlayFabSettings::AD_TYPE_IDFA)
-                    request.Idfa = apiSettings->advertisingIdValue;
-                else if (apiSettings->advertisingIdType == PlayFabSettings::AD_TYPE_ANDROID_ID)
-                    request.Adid = apiSettings->advertisingIdValue;
-                else
-                    return;
-                AttributeInstall(request, nullptr, nullptr);
-            }
+            AttributeInstallRequest request;
+            if (context->advertisingIdType == PlayFabSettings::AD_TYPE_IDFA)
```

Comment:
please use braces - here and other places .

Reply:
[https://github.com/PlayFab/XPlatCoreTemplate/pull/40](https://github.com/PlayFab/XPlatCoreTemplate/pull/40) I will fix this issue in a later PR.

---
In reply to: [342834511](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r342834511) [](ancestors = 342834511)

Reply:
Appropriate braces added to everything I touched.

---
In reply to: [343240742](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343240742) [](ancestors = 343240742,342834511)